src/solvers/conjugate_gradient_descent.jl

```diff
@@ -107,7 +107,7 @@ function conjugate_gradient_descent!(
         p,
         stopping_criterion,
         stepsize,
-        coefficient,
+        DirectionUpdateRuleStorage(M, coefficient, p, X),
```

Comment:
as written above in the CG constructor - this could also be done in the CG constructor maybe, because then one does not have to change all the tests below :)